lib/controller.rb

```diff
@@ -4,7 +4,22 @@ def initialize(game)
   end
 
   def play
-    @game.play_move until @game.is_over?
-    @game.tie_or_won
+    play_game = true
+    while play_game
+      @game.play_move until @game.is_over?
+      @game.tie_or_won
+      play_game = play_again?
+    end
+    @game.display.show_exit_message
```

Comment:
This works, and shows you understand how to solve the problem. Which is the most important thing... but the play method now has two nested loop the `play_game` loop and the `play_move` loop. How do you test this? My suggestion is that these should be different objects not just different methods.